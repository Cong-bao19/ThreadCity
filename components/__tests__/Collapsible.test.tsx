import React from 'react';
import { useColorScheme } from '@/hooks/useColorScheme';
import renderer from 'react-test-renderer';
import { Collapsible } from '../Collapsible';

// Mock useColorScheme hook
jest.mock('@/hooks/useColorScheme', () => ({
  useColorScheme: jest.fn(),
}));

describe('Collapsible', () => {
  it('renders closed by default', () => {
    // Mock trả về theme là 'light'
    (useColorScheme as jest.Mock).mockReturnValue('light');  // Type casting để đảm bảo jest.Mock được nhận diện

    const tree = renderer.create(
      <Collapsible title="Test Title">Content</Collapsible>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
