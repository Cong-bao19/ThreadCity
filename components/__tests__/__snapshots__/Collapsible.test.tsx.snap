// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible renders closed by default 1`] = `"Test Title"`;
